azurelinuxagent/ga/exthandlers.py

```diff
@@ -708,6 +718,70 @@ def report_event(self, message="", is_success=True, duration=0):
         add_event(name=self.ext_handler.name, version=ext_handler_version, message=message,
                   op=self.operation, is_success=is_success, duration=duration)
 
+    def try_get_pkg_file(self):
```

Comment:
We have a handle to this file when we unzip it. Why not either:

- remove the zip right after we successfully unzip
- persist that reference until the point you want to remove the zip

I vote for the first option.

Reply:
Are there patterns of recovery from install failures where, today, we'd reuse the zip already on the disk? Or does the agent always download a fresh copy of the zip?

Reply:
I have the same concerns that Jason does, which is why I opted for this path.  I **thought** there were situations were the agent would continually download something in the case of failure, and I wanted to avoid that unless there was an explicit uninstall.